tests/Unit/Hooks/GuzzleMiddlewareTest.php

```diff
@@ -19,8 +19,8 @@
 
     $middleware = new GuzzleMiddleware(nightwatch());
 
-    $stack = $middleware(fn () => new Response(body: 'ok'));
-    $response = $stack(new Request('GET', '/test'), []);
+    $stack = $middleware(fn () => new FulfilledPromise(new Response(body: 'ok')));
+    $response = $stack(new Request('GET', '/test'), [])->wait();
```

Comment:
This was using bad types. Adding the return type to the middleware exposed that this was wrong.